File: camerax_android/android/src/main/kotlin/dev/hebei/camerax_android/view/video/AudioConfig.kt
package dev.hebei.camerax_android.view.video

import android.Manifest
import androidx.annotation.RequiresPermission

class AudioConfig internal constructor(internal val obj: androidx.camera.view.video.AudioConfig) {
    companion object {
        val AUDIO_DISABLED get() = AudioConfig(androidx.camera.view.video.AudioConfig.AUDIO_DISABLED)

        @RequiresPermission(Manifest.permission.RECORD_AUDIO)
        fun create(enableAudio: Boolean): AudioConfig {
            val obj = androidx.camera.view.video.AudioConfig.create(enableAudio)
            return AudioConfig(obj)
        }
    }

    val audioEnabled get() = obj.audioEnabled
}

File: camerax_ios/ios/Classes/view/video/AudioConfig.swift
//
//  AudioConfig.swift
//  camerax_ios
//
//  Created by 闫守旺 on 2025/5/15.
//

import Foundation

public class AudioConfig: NSObject {
    static let audioDisabled = AudioConfig.create(false)
    
    static func create(_ enableAudio: Bool) -> AudioConfig {
        return AudioConfig(enableAudio)
    }
    
    public let audioEnabled: Bool
    
    private init(_ audioEnabled: Bool) {
        self.audioEnabled = audioEnabled
    }
}
